SPI: Fix deprecated API usage leftovers from #87427

jhedberg opened the pull request:
#87427 caused some [CI failures in main](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18160982496/job/51691443328) due some drivers still using the SPI macros in a deprecated way, and actually `SPI_DT_IODEV_DEFINE()` not having the deprecation warning support. This PR tries to address both issues.

jhedberg committed:
include: drivers: spi.h: Fix deprecation support for SPI_DT_IODEV_DEFINE

Commit a60f93d742a48660882af5785fb4352f414ecd11 removed the delay parameter
from various SPI macros, but aimed to retain a clean migration path by
throwing a deprecation warning when the parameter is still passed. However,
the commit failed to have this feature included in the SPI_DT_IODEV_DEFINE
macro. This commit should fix the issue.

Signed-off-by: Johan Hedberg <johan.hedberg@silabs.com>

jhedberg committed:
drivers: sensor: Fix deprecated SPI_DT_IODEV_DEFINE usage

The delay parameter was removed from this macro, so don't pass it.

Signed-off-by: Johan Hedberg <johan.hedberg@silabs.com>

decsny reviewed (dismissed):
yeah sorry, it was taking forever to get reviews and stuff kept merging in the meantime, i couldnt keep up. looks like this driver is newly added yesterday right before the spi PR was merged

jhedberg committed:
drivers: Fix deprecated SPI_DT_SPEC_INST_GET usage

This macro doesn't accept a delay parameter anymore (well, it does, but
it's deprecated and will trigger build warnings). Just remove it from the
places that were still passing it.

Signed-off-by: Johan Hedberg <johan.hedberg@silabs.com>

sylvioalves commented:
@jhedberg it is good now.
